app/forms/idv/api_image_upload_form.rb

```diff
@@ -121,6 +124,20 @@ def validate_pii_from_doc(client_response)
       response
     end
 
+    def doc_type_and_issuing_country(pii_from_doc)
+      front_issuing_country = pii_from_doc&.dig(:Front, 'CountryCode')
+      back_issuing_country = pii_from_doc&.dig(:Back, 'CountryCode')
+      front_doc_type = pii_from_doc&.dig(:Front, 'ClassName')
+      back_doc_type = pii_from_doc&.dig(:Back, 'ClassName')
```

Comment:
it's surprising to me to see symbol and strings as hash keys mixed like this? what causes the source data to be like this?

Reply:
Thank you for bringing this up. It's the structure I saw when `binding.pry` and looking at the data coming through, but having looked at a real-world response it looks like both the keys should both be strings. I think the mock response process here is doing some conversion.

I'm looking into how to change it.

Reply:
@zachmargolis, the data is from parsing response from acuant/ln, 

https://github.com/18F/identity-idp/blob/83eeab2a4e3bf937202e5c53dcbd34d45f8c1d51/app/services/doc_auth/acuant/responses/get_results_response.rb#L110-L127

The data was merged to response hash directly. 

My opinion is to create a new card to deal with it, since we need to manage it on both acuant/lexisnexis responses.

Reply:
this tells me where the `CamelCase` string names come from, but it does not me where the `:Camel` symbol names came from? I'm only seeing `:front` and `:back` in this file, not `:Front` and `:Back`?

Reply:
@zachmargolis ,  the following is the hash template this method returns.

```
{"Back"=>{"ClassName"=>"Unknown", "Issue"=>nil, "IssueType"=>nil, "Name"=>"Unknown", "IssuerCode"=>nil, "IssuerName"=>nil, "CountryCode"=>""}, "Front"=>{"ClassName"=>"Unknown", "Issue"=>nil, "IssueType"=>nil, "Name"=>"Unknown", "IssuerCode"=>nil, "IssuerName"=>nil, "CountryCode"=>""}}
```

The result is assembled in this method of the class
https://github.com/18F/identity-idp/blob/83eeab2a4e3bf937202e5c53dcbd34d45f8c1d51/app/services/doc_auth/acuant/responses/get_results_response.rb#L54-L71

Reply:
Thank you both for looking into this. As I mentioned in my previous comment, I had a suspicion that maybe they should all be strings. This is because when I looked a copy of what's actually returned from Acuant, I saw all of the keys were strings.

But, as I did more research, I found more of the `pii` keys were symbols, despite also being strings in that document.

Further research told me we're converting these keys to strings [in True ID responses](https://github.com/18F/identity-idp/blob/31d4f095c1a4fd6b1fbdbf4fe46c5788587cce94/app/services/doc_auth/lexis_nexis/responses/true_id_response.rb#L89) with `classification_info` [defined as a hash with symbol keys here](https://github.com/18F/identity-idp/blob/31d4f095c1a4fd6b1fbdbf4fe46c5788587cce94/app/services/doc_auth/lexis_nexis/responses/true_id_response.rb#L192).

Acuant was a little harder to clarify, but [in Acuant responses](https://github.com/18F/identity-idp/blob/31d4f095c1a4fd6b1fbdbf4fe46c5788587cce94/app/services/doc_auth/acuant/pii_from_doc.rb#L29) we're also converting the top keys to hashes. After some experimentation in `irb`, I believe [what Dawei linked above](https://github.com/18F/identity-idp/blob/83eeab2a4e3bf937202e5c53dcbd34d45f8c1d51/app/services/doc_auth/acuant/responses/get_results_response.rb#L110-L127)  also converts the keys to hashes surreptitiously ([see my comment in Slack here](https://gsa-tts.slack.com/archives/C05HSH9RQ57/p1697839795510729)).

I pushed up a commit ([28fe49c](https://github.com/18F/identity-idp/pull/9427/commits/28fe49cec096d91904d0621f12b654991a39d59a)) to convert to symbols in the mock as well, and fix the tests.

I'm not 100% sure I'm right about this, so I welcome any feedback.